### DISCUSSION TITLE
Color looks off and washed out

### DISCUSSION THREAD

**Discussion by lohant** [LAutour/ESP32-HUB75-MatrixPanel-DMA-ICN2053#3] [Q&A]
Hi,

I have a problem with the colors appearing to look washed out and at a lower brightness. Here is an image that displays the problem: https://imgtr.ee/i/ptVYB
The problem is even more noticable irl.

To the left is a display running the library by mrfaptastic and to the right is a display running this library.  Both setups run the same code except that the library has been switched out and "flipDMABuffer();" has been added twice after the image is generated in the code using this library. I noticed that some functions regarding colors of text, etc, did not work right away when I copied the code I used with the other library. Thus, I suspect that the author of this library has made som changes to how the colors work. As I am a beginner in C++, I would greatly appreciate if someone could point me in the right direction.

Thanks!

**Comment by LAutour** [comment]
The based color conversion palette is set in color-convert.cpp
lumConvTab[256]
For my projects, I used the simplest graphics, so I did not check the display of full-color images. Plus, I focused on uniform color brightness for anti-aliased lines (writeLineAA) .

**Reply by LAutour** [reply]
Recalculation of the current palette based on lumConvTab
done in ESP32-HUB75-MatrixPanel-DMA-lowdraw.cpp setPanelBrightness
The recalculation is approximate - you can change it.

**Reply by lohant** [reply]
Ok. After looking into it, I understand the logic behind the brightness. I do not understand where I should adjust the colors though? I am using drawRBBBitmap, do you have any suggestion?

**Reply by lohant** [reply]
To be clear, I'd like it to display non-adjusted colors. Just like the other library does.

**Reply by LAutour** [reply]
wait for the library update - I'll try to redo something

**Reply by lohant** [reply]
Ok. I really appreciate your work on this library. Do you have any idea when the library may be updated?

**Reply by LAutour** [reply]
In progress. They dragged me electronics for repairs - I'm busy with it.
